hat/app/org/hatdex/hat/utils/NamespaceUtils.scala

```diff
@@ -46,4 +47,55 @@ object NamespaceUtils {
     }
     matchedRoles.flatten.nonEmpty
   }
+
+  def getWriteNamespace(roles: Seq[UserRole]): Option[String] = {
+    val writeNamespaces = roles.map {
+      case NamespaceWrite(n) => n
+      case _                 => ""
+    }
+
+    if (writeNamespaces.isEmpty)
+      None
+    else
+      Some(writeNamespaces.head)
```

Comment:
this could be `Some("")` which I would say is a bug, also the whole method could be written much more simply and correctly:
`roles.collectFirst { case NamespaceWrite(ns) => ns }`

this is only correct of course if there is at most one write namespace